lib/http-invocation.js

```diff
@@ -223,14 +228,30 @@ HttpInvocation.prototype.invoke = function(callback) {
   if (!this.req) {
     this.createRequest();
   }
-  request(this.req, function(err, res, body) {
-    if (err instanceof SyntaxError) {
-      if (res.status === 204) err = null;
+  var method = this.method;
+  var streamDesc = method.sharedMethod.stream;
+  var methodReturnsStream = !!streamDesc;
+  var streamResult;
+
+  if (methodReturnsStream) {
+    if (streamDesc.type === 'ReadableStream' && streamDesc.json) {
+      streamResult = request(this.req).pipe(JSONStream.parse());
+      callback(null, streamResult);
+    } else {
+      callback(new Error('unsupported stream type'));
```

Comment:
Please add more info to the error message, make it easier to troubleshoot the problem.